### PR TITLE
[New] Query feature table

### PR DESCRIPTION
**Pull request by pgruenler**
## Description

This PR implements `Query feature table` in `Features` category.
URL to README: 

## Linked Issue(s)

- `swift/issues/3577`

### CODE REVIEW

File: Shared/Samples/Feature layer query/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Feature layer query
```
Comment (yo1995) [inline Shared/Samples/Feature layer query/README.md:1]: Hi Phil, can you help rename the sample to "Query feature table" according to the common design? There was an effort to standardize the names across platforms, and the old iOS sample name deviates from the common design. Thanks.

- the folder name
- README title
- title in metadata
- swift filename
- view type name (FeatureLayerQueryView -> QueryFeatureTableView)

Comment (pgruenler) [reply Shared/Samples/Feature layer query/README.md:1]: Ah, I saw there were different names in the design and the old samples; I picked the wrong one. I'll make the changes.

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    class Model: ObservableObject {
+        /// A map with a topographic basemap style.
+        let map = Map(basemapStyle: .arcGISTopographic)
+        
+        @Published var currentQuery = ""
+        
+        var selectedFeatures: [Feature] = []
+        
+        // Create a feature table using a url.
+        let featureTable = ServiceFeatureTable(url: URL(string: "https://services.arcgis.com/jIL9msH9OI208GCb/arcgis/rest/services/USA_Daytime_Population_2016/FeatureServer/0")!)
```
Comment (des12437) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:107]:
```suggestion
        let featureTable = ServiceFeatureTable(
             url: URL(string: "https://services.arcgis.com/jIL9msH9OI208GCb/arcgis/rest/services/USA_Daytime_Population_2016/FeatureServer/0")!
)
```

**Review by des12437 — COMMENTED** [review]
A few whitespace suggestions.

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        let map = Map(basemapStyle: .arcGISTopographic)
+        
+        @Published var currentQuery = ""
+        
+        var selectedFeatures: [Feature] = []
+        
+        // Create a feature table using a url.
+        let featureTable = ServiceFeatureTable(
+             url: URL(string: "https://services.arcgis.com/jIL9msH9OI208GCb/arcgis/rest/services/USA_Daytime_Population_2016/FeatureServer/0")!
+)
```
Comment (des12437) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:108]:
```suggestion
        )
```
The indentation was off in my first suggestion.

**Review by des12437 — DISMISSED** [review]

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                    // Unselect all selected features.
+                    featureLayer.unselectFeatures(model.selectedFeatures)
+                    model.selectedFeatures.removeAll()
+                    // Make the query parameters and execute the query.
+                    let queryParameters = QueryParameters()
+                    queryParameters.whereClause = "upper(STATE_NAME) LIKE '%\(query.uppercased())%'"
+                    do {
+                        let queryResult = try await model.featureTable.queryFeatures(using: queryParameters)
+                        model.selectedFeatures = Array(queryResult.features())
+                        if !model.selectedFeatures.isEmpty {
```
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:73]:
```suggestion
                        let queryResultFeatures = Array(queryResult.features())
                        if !queryResultFeatures.isEmpty {
```

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                        }
+                    }
+                    .padding()
+                }
+                .task(id: query) {
+                    // Makes sure we have a query string and a layer.
+                    guard !query.isEmpty, let featureLayer = model.featureLayer else { return }
+                    // Unselect all selected features.
+                    featureLayer.unselectFeatures(model.selectedFeatures)
+                    model.selectedFeatures.removeAll()
```
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:66]:
```suggestion
                    // Unselects all features.
                    model.featureLayer.clearSelection()
```

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+                    // Make the query parameters and execute the query.
+                    let queryParameters = QueryParameters()
+                    queryParameters.whereClause = "upper(STATE_NAME) LIKE '%\(query.uppercased())%'"
+                    do {
+                        let queryResult = try await model.featureTable.queryFeatures(using: queryParameters)
+                        model.selectedFeatures = Array(queryResult.features())
+                        if !model.selectedFeatures.isEmpty {
+                            // Display the selection.
```
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:74]:
```suggestion
                            // Displays the selection.
```

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                    }
+                    .padding()
+                }
+                .task(id: query) {
+                    // Makes sure we have a query string and a layer.
+                    guard !query.isEmpty, let featureLayer = model.featureLayer else { return }
+                    // Unselect all selected features.
+                    featureLayer.unselectFeatures(model.selectedFeatures)
+                    model.selectedFeatures.removeAll()
+                    // Make the query parameters and execute the query.
```
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:67]:
```suggestion
                    // Makes the query parameters and executes the query.
```

**Comment by des12437** [comment]
Please resolve the merge conflicts.

File: Shared/Samples/Query feature table/QueryFeatureTableView.swift [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(map: model.map)
+                .alert(isPresented: $isShowingAlert, presentingError: error)
+                // Makes the search bar.
+                .searchable(text: $searchBarText, prompt: Text("Search state names"))
+                .onSubmit(of: .search, {
+                    model.currentQuery = searchBarText
+                })
```
Comment (yo1995) [inline Shared/Samples/Query feature table/QueryFeatureTableView.swift:41]: I was a bit surprised that the [trailing closure linter rule](https://realm.github.io/SwiftLint/trailing_closure.html) isn't triggered on this line. So it's up to you if make this change.

```suggestion
                .onSubmit(of: .search) {
                    model.currentQuery = searchBarText
                }
```

**Review by yo1995 — COMMENTED** [review]
Will approve once the merge conflict is resolved.

**Review by des12437 — APPROVED** [review]

**Review by yo1995 — APPROVED** [review]

File: Samples.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -180,6 +182,10 @@
 				883C121929C914E100062FF9 /* DownloadPreplannedMapAreaView.swift in Copy Source Code Files */,
 				1C0C1C3D29D34DDD005C8B24 /* ChangeViewpointView.swift in Copy Source Code Files */,
 				1C42E04A29D239D2004FC4BE /* ShowPopupView.swift in Copy Source Code Files */,
+				108EC04329D25D6D000F35D0 /* ChangeMapViewBackgroundView.swift in Copy Source Code Files */,
+				108EC04429D25D6D000F35D0 /* ChangeMapViewBackgroundView.SettingsView.swift in Copy Source Code Files */,
+				108EC04529D25D6D000F35D0 /* ChangeMapViewBackgroundView.Model.swift in Copy Source Code Files */,
```
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:187]: These changes shouldn't be here - will be fixed in another PR.

**Pull request merged**